ts/util/i18n/localizedString.ts

```diff
@@ -0,0 +1,303 @@
+      ? Var | StringArgs<Rest>
+      : never;
+
+export type StringArgsRecord<T extends string> = Record<StringArgs<T>, string | number>;
+
+// TODO: move this to a test file
+//
+// const stringArgsTestStrings = {
+//   none: 'test',
+//   one: 'test{count}',
+//   two: 'test {count} second {another}',
+//   three: 'test {count} second {another} third {third}',
+//   four: 'test {count} second {another} third {third} fourth {fourth}',
+//   five: 'test {count} second {another} third {third} fourth {fourth} fifth {fifth}',
+//   twoConnected: 'test {count}{another}',
+//   threeConnected: '{count}{another}{third}',
+// } as const;
+//
+// const stringArgsTestResults = {
+//   one: { count: 'count' },
+//   two: { count: 'count', another: 'another' },
+//   three: { count: 'count', another: 'another', third: 'third' },
+//   four: { count: 'count', another: 'another', third: 'third', fourth: 'fourth' },
+//   five: { count: 'count', another: 'another', third: 'third', fourth: 'fourth', fifth: 'fifth' },
+//   twoConnected: { count: 'count', another: 'another' },
+//   threeConnected: { count: 'count', another: 'another', third: 'third' },
+// } as const;
+//
+// let st0: Record<StringArgs<typeof stringArgsTestStrings.none>, string>;
+// const st1: Record<StringArgs<typeof stringArgsTestStrings.one>, string> = stringArgsTestResults.one;
+// const st2: Record<StringArgs<typeof stringArgsTestStrings.two>, string> = stringArgsTestResults.two;
+// const st3: Record<
+//   StringArgs<typeof stringArgsTestStrings.three>,
+//   string
+// > = stringArgsTestResults.three;
+// const st4: Record<
+//   StringArgs<typeof stringArgsTestStrings.four>,
+//   string
+// > = stringArgsTestResults.four;
+// const st5: Record<
+//   StringArgs<typeof stringArgsTestStrings.five>,
+//   string
+// > = stringArgsTestResults.five;
+// const st6: Record<
+//   StringArgs<typeof stringArgsTestStrings.twoConnected>,
+//   string
+// > = stringArgsTestResults.twoConnected;
+// const st7: Record<
+//   StringArgs<typeof stringArgsTestStrings.threeConnected>,
+//   string
+// > = stringArgsTestResults.threeConnected;
+//
+// const results = [st0, st1, st2, st3, st4, st5, st6, st7];
+
+// Above is testing stuff
+
+function getPluralKey<R extends PluralKey>(string: PluralString): R {
+  const match = /{(\w+), plural, one \[.+\] other \[.+\]}/g.exec(string);
+  return match?.[1] as R;
+}
+
+function getStringForCardinalRule(
+  localizedString: string,
+  cardinalRule: Intl.LDMLPluralRule
+): string | undefined {
+  // TODO: investigate if this is the best way to handle regex like this
+  const cardinalPluralRegex: Record<Intl.LDMLPluralRule, RegExp> = {
+    zero: /zero \[(.*?)\]/g,
+    one: /one \[(.*?)\]/g,
+    two: /two \[(.*?)\]/g,
+    few: /few \[(.*?)\]/g,
+    many: /many \[(.*?)\]/g,
+    other: /other \[(.*?)\]/g,
+  };
+  const regex = cardinalPluralRegex[cardinalRule];
+  const match = regex.exec(localizedString);
+  return match?.[1] ?? undefined;
+}
```

Comment:
this is duplicated between two files, to cleanup `getStringForCardinalRule` (maybe others too)

Reply:
fixed